lib/core/src/text.rs

```diff
@@ -0,0 +1,229 @@
+use web_sys::{WebGlRenderingContext,WebGlBuffer,WebGlTexture};
+
+pub struct TextComponentBuilder<'a> {
+    pub text     : String,
+    pub font     : &'a mut FontRenderInfo,
+    pub position : Vector2<f64>,
+    pub size     : f64,
+    pub color    : Color<f32>,
+}
+
+#[derive(Debug)]
+pub struct TextComponent {
+    gl_context       : WebGlRenderingContext,
+    gl_program       : Program,
+    gl_vertex_buf    : WebGlBuffer,
+    gl_tex_coord_buf : WebGlBuffer,
+    gl_msdf_texture  : WebGlTexture,
+    buffers_size     : usize,
+}
+
+impl<'a> TextComponentBuilder<'a> {
+    pub fn build(mut self, workspace  : &Workspace) -> TextComponent {
+        self.load_all_chars();
+        let gl_context           = workspace.context.clone();
+        let gl_program           = self.create_program(&gl_context);
+        let gl_vertex_buf        = self.create_vertex_buf(&gl_context);
+        let gl_tex_coord_buf     = self.create_tex_coord_buf(&gl_context);
+        let gl_msdf_texture      = self.create_msdf_texture(&gl_context);
+        let glyph_vertices_count = glyph_render::GLYPH_SQUARE_VERTICES_BASE_LAYOUT.len();
+        let buffers_size         = self.text.len() * glyph_vertices_count;
+
+        self.setup_uniforms(&gl_context, &gl_program);
+
+        TextComponent
+        {gl_context,gl_program,gl_vertex_buf,gl_tex_coord_buf,gl_msdf_texture,buffers_size}
+    }
+
+    fn load_all_chars(&mut self) {
+        for ch in self.text.chars() {
+            self.font.get_glyph_info(ch);
+        }
+    }
+
+    fn create_program(&self, gl_context:&Context) -> Program {
+        gl_context.get_extension("OES_standard_derivatives").unwrap().unwrap();
+
+        let vert_shader = self.create_vertex_shader(gl_context);
+        let frag_shader = self.create_fragment_shader(gl_context);
+
+        link_program(&gl_context, &vert_shader, &frag_shader).unwrap()
+    }
+
+    fn create_vertex_shader(&self, gl_context:&Context) -> Shader {
+        let body        = include_str!("text/msdf_vert.glsl");
+        let shader_type = WebGlRenderingContext::VERTEX_SHADER;
+
+        compile_shader(gl_context,shader_type,body).unwrap()
+    }
+
+    fn create_fragment_shader(&self, gl_context:&Context) -> Shader {
+        let body        = include_str!("text/msdf_frag.glsl");
+        let shader_type = WebGlRenderingContext::FRAGMENT_SHADER;
+
+        compile_shader(gl_context,shader_type,body).unwrap()
+    }
+
+    fn create_buffer(gl_context:&Context, vertices:&[f32]) -> WebGlBuffer {
+        let buffer = gl_context.create_buffer().unwrap();
+        let target = WebGlRenderingContext::ARRAY_BUFFER;
+        let usage  = WebGlRenderingContext::STATIC_DRAW;
+        gl_context.bind_buffer(target,Some(&buffer));
+
+        unsafe { // Note [unsafe buffer_data]
+            let float_32_array = Float32Array::view(&vertices);
+            gl_context.buffer_data_with_array_buffer_view(target,&float_32_array,usage);
+        }
```

Comment:
Do you think we could refactor this lines to a separate safe function with a safety note? I think it would be much more clear.